lib/semantics/check-do.cc

```diff
@@ -219,6 +214,46 @@ class DoConcurrentBodyEnforce {
   }
 
 private:
+  void CheckAllocateObjectIsCoarray(
+      const parser::AllocateObject &allocateObject, const char *stmtType) {
+    const Symbol *symbol{nullptr};
+    parser::CharBlock position;
+    std::visit(
+        common::visitors{
+            [&](const parser::Name &x) {
+              symbol = x.symbol;
+              position = x.source;
+            },
+            [&](const parser::StructureComponent &x) {
+              symbol = x.component.symbol;
+              position = x.component.source;
+            },
+        },
+        allocateObject.u);
+    if (IsCoarray(*symbol)) {
+      context_.Say(position,
+          "%s coarray not allowed in DO CONCURRENT"_err_en_US, stmtType);
+    }
+  }
+
+  void CheckAllocateContainsCoarray(const parser::AllocateStmt &allocateStmt) {
```

Comment:
The name of this function suggests it's doing the opposite of what it actually does. Maybe `CheckDoesntContainCoarray()`?

Reply:
Thanks.  I've changed the names.